### PR TITLE
Modify rule S6317: Add message, highlighting and the list of supported privilege escalation vector details

### PR DESCRIPTION
**Pull request by pierre-loup-tristant-sonarsource**
Ticket: [SONARIAC-384](https://jira.sonarsource.com/browse/SONARIAC-384)

### CODE REVIEW

File: rules/S6317/message.adoc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+=== Message
+
+* Main location
+** This policy is vulnerable to the "{vector_name}" privilege escalation vector. Remove permissions or restrict the set of resources they apply to.
+* Secondary locations
+** This permission enables the "{vector_name}" escalation vector.
+** When combined with others, this permission enables the "{vector_name}" escalation vector.
+** Permissions are granted on all resources
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6317/message.adoc:8]:
```suggestion
** Permissions are granted on all resources.
```

File: rules/S6317/description.adoc [hunk 2]
```diff
@@ -48,6 +48,77 @@ Although it looks like it grants a limited set of permissions, this policy would
...
+|Create a new access key for any user
+
+|Create Login Profile
+|Create an access with a chosen password
+
+|Update Login Profile 
+|Update the existing password
+
+|Attach User Policy
+|Attach a permissive IAM policy like "AdministratorAccess" to a user the attacker contols
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6317/description.adoc:73]:
```suggestion
|Attach a permissive IAM policy like "AdministratorAccess" to a user the attacker controls
```

File: rules/S6317/description.adoc [hunk 3]
```diff
@@ -48,6 +48,77 @@ Although it looks like it grants a limited set of permissions, this policy would
...
+
+|Attach Group Policy
+|Attach a permissive IAM policy like "AdministratorAccess" to a group containing a user the attacker contols
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6317/description.adoc:76]:
```suggestion
|Attach a permissive IAM policy like "AdministratorAccess" to a group containing a user the attacker controls
```

File: rules/S6317/description.adoc [hunk 4]
```diff
@@ -48,6 +48,77 @@ Although it looks like it grants a limited set of permissions, this policy would
...
+
+|Attach Role Policy
+|Attach a permissive IAM policy like "AdministratorAccess" to a role that can be assumed by the user the attacker contols
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6317/description.adoc:79]:
```suggestion
|Attach a permissive IAM policy like "AdministratorAccess" to a role that can be assumed by the user the attacker controls
```

File: rules/S6317/description.adoc [hunk 5]
```diff
@@ -48,6 +48,77 @@ Although it looks like it grants a limited set of permissions, this policy would
...
+
+|Put User Policy
+|Alter the existing inline IAM policy from a user the attacker contols
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6317/description.adoc:82]:
```suggestion
|Alter the existing inline IAM policy from a user the attacker controls
```

File: rules/S6317/description.adoc [hunk 6]
```diff
@@ -48,6 +48,77 @@ Although it looks like it grants a limited set of permissions, this policy would
...
+
+|Put Group Policy
+|Alter the existing inline IAM policy from a group containing a user that the attacker contols
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6317/description.adoc:85]:
```suggestion
|Alter the existing inline IAM policy from a group containing a user that the attacker controls
```

File: rules/S6317/description.adoc [hunk 7]
```diff
@@ -48,6 +48,77 @@ Although it looks like it grants a limited set of permissions, this policy would
...
+
+|Put Role Policy
+|Alter an existing inline IAM role policy. The rule will then be assumed by the user that the attacker contols
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6317/description.adoc:88]:
```suggestion
|Alter an existing inline IAM role policy. The rule will then be assumed by the user that the attacker controls
```

File: rules/S6317/description.adoc [hunk 8]
```diff
@@ -48,6 +48,77 @@ Although it looks like it grants a limited set of permissions, this policy would
...
+
+|Add User to Group
+|Add a user that the attacker contols to a group that has a larger range of permissions
+
+|Update Assume Role Policy
+|Update a role's "AssumeRolePolicyDocument" to allow a user the attacker contols to assume it
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6317/description.adoc:94]:
```suggestion
|Update a role's "AssumeRolePolicyDocument" to allow a user the attacker controls to assume it
```

File: rules/S6317/description.adoc [hunk 9]
```diff
@@ -48,6 +48,77 @@ Although it looks like it grants a limited set of permissions, this policy would
...
+|Alter the existing inline IAM policy from a user the attacker contols
+
+|Put Group Policy
+|Alter the existing inline IAM policy from a group containing a user that the attacker contols
+
+|Put Role Policy
+|Alter an existing inline IAM role policy. The rule will then be assumed by the user that the attacker contols
+
+|Add User to Group
+|Add a user that the attacker contols to a group that has a larger range of permissions
```
Comment (hendrik-buchwald-sonarsource) [inline rules/S6317/description.adoc:91]:
```suggestion
|Add a user that the attacker controls to a group that has a larger range of permissions
```

**Review by hendrik-buchwald-sonarsource — APPROVED** [review]
LGTM, just some small typos.

**Comment by loris-s-sonarsource** [comment]
⚠️ I merged this by mistake

**Pull request merged**